robots/preservation_ingest/update_catalog.rb

```diff
@@ -21,7 +21,21 @@ def perform(druid)
         private
 
         def update_catalog
-          # FIXME:  do something and update the comment too
+          size = moab_object.size
+          storage_location = moab_object.storage_root
+          latest_version_id = moab_object.current_version_id
+          args = {
+            druid: druid,
+            incoming_version: latest_version_id,
+            incoming_size: size,
+            storage_location: storage_location
+          }
+          response = conn.post '/catalog', args
```

Comment:
It seems to me that you can check the latest_version_id and KNOW if the object is new or not, thus avoiding extra HTTP calls.

In other words, if it's version 1, it's a new object and we do a POST;  if it's not version 1, then we do a PATCH.

If either of these fails (give or take retries), then it's an error -- we know we should be able to create in pres cat if it's a new object, and we know we should be able to update in pres cat if it's a new version.